conftest.py

```diff
@@ -1,101 +1,63 @@
 import logging
-import os
 
+from leapp.repository.scan import find_and_scan_repositories
+from leapp.utils.repository import find_repository_basedir
 
 logger = logging.getLogger(__name__)
-
-
-def load_repo(path):
-    """
-    Load repository on demand.
-
-    Do not require paths initialized if no environment is set.
-    Allows some parts to be tested without working leapp installation.
-    """
-    from leapp.utils.repository import find_repository_basedir
-    from leapp.repository.scan import find_and_scan_repositories
-
-    repo = find_and_scan_repositories(
-        find_repository_basedir(path), include_locals=True
-    )
-    repo.load()
-    return repo
+logging.getLogger("asyncio").setLevel(logging.INFO)
+logging.getLogger("parso").setLevel(logging.INFO)
 
 
 def pytest_collectstart(collector):
     if collector.nodeid:
-        if not hasattr(collector.session, "leapp_repository"):
-            repo = load_repo(collector.nodeid)
+        current_repo_basedir = find_repository_basedir(collector.nodeid)
+        # loading the current repo
+        if (
+            not hasattr(collector.session, "leapp_repository")
+            or current_repo_basedir != collector.session.repo_base_dir
+        ):
+            repo = find_and_scan_repositories(
+                find_repository_basedir(collector.nodeid), include_locals=True
+            )
+            repo.load()
             collector.session.leapp_repository = repo
-
-        actor = None
-        # find which actor is being tested
-        for a in collector.session.leapp_repository.actors:
-            if a.full_path == collector.fspath.dirpath().dirname:
-                actor = a
-                break
-
-        if not actor:
-            logger.info("No actor found, exiting collection...")
-            return
-
-        # load actor context so libraries can be imported on module level
-
-        logger.info(
-            "Injecting actor context for {}".format(repr(actor.class_name))
-        )
-        collector.actor_context = actor.injected_context()
-        collector.actor_context.__enter__()
-        logger.info("Actor context injected")
-
-
-#
-# def pytest_itemcollected(item):
-#     item
-#
-# def pytest_internalerror(excrepr, excinfo):
-#     excrepr, excinfo
-#
-# def pytest_exception_interact(node, call, report):
-#     node, call, report
-#
-# def pytest_collection_modifyitems(session, config, items):
-#     session, config, items
-#
-# def pytest_pycollect_makemodule(path, parent):
-#     path, parent
-#
-#
-# def pytest_pycollect_makeitem(collector, name, obj):
-#     collector
-#
-# def pytest_generate_tests(metafunc):
-#     metafunc
-#
```

Comment:
In the final version, do not forget to remove not add this code when you are removing it here. Not sure what will be the final set of commits. Just want to ensure that these comments will be added in one commit and removed in another during the one PR. Obviously this could be resolved just by squash of two commits.